### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by shuhad786**
# Create a class Classroom with the following:
- @label instance variable, should be initialized in the constructor.
- Setter and getter for @label (remember about attr_accessor).
- Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
- Create the has-many side (a classroom has many students).
- Create the belongs-to side (a student belongs to a classroom).
- Make sure that when adding a student to a classroom it also sets the classroom for the student.
- Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
# Create a class Book with the following:
- @title and @author instance variables, should be initialized in the constructor.
- Setters and getters for instance variables (remember about attr_accessor).
- Create a class Rental with the following:
- @date instance variable, should be initialized in the constructor.
- Setter and getter for @date (remember about attr_accessor).
# Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
- Create the has-many side of Book and Rental (a book has many rentals).
- Create the belongs-to side of Rental and Book (a rental belongs to a book).
- Create the has-many side of Person and Rental (a person has many rentals).
- Create the belongs-to side of Rental and Person (a rental belongs to a person).
- Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
```
Comment (Nemwel-Boniface) [inline rental.rb]: - [ ] Kindly add the book and person to the getter and setter (attr_accessor). This is how it is supposed to be ```attr_accessor :date, :book, :person```

File: person.rb [hunk 2]
```diff
@@ -1,14 +1,19 @@
 require './nameable'
+require './capitalize_decorator'
+require './trimmer_decorator'
+require './rental'
 
 class Person < Nameable
-  attr_accessor :id, :name, :age
+  attr_accessor :name, :age
```
Comment (Nemwel-Boniface) [inline person.rb]: - [ ] You might have forgotten to add the rental to the attribute accessor. Kindly add it.

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @shuhad786          :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Good job working on your project so far! :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by gemmen29 — APPROVED** [review]
Hi @shuhad786, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**